models.py

```diff
@@ -0,0 +1,254 @@
+    with open(dest, "wb") as dest_file:
+        with tqdm(
+            desc="Downloading model",
+            total=total_size,
+            unit="iB",
+            unit_scale=True,
+            unit_divisor=1024,
+        ) as progress:
+            for data in resp.iter_content(chunk_size):
+                dest_file.write(data)
+                progress.update(len(data))
+
+
+def get_downloaded_model_path(model_name: str) -> Path:
+    """Fetch the segment-anything model to a local cache directory."""
+    model_url = MODEL_URLS[model_name]
+
+    model_location = MODEL_DIR / model_url.split("/")[-1]
+    if not model_location.exists():
+        os.makedirs(MODEL_DIR, exist_ok=True)
+        download_with_progress(model_url, model_location)
+
+    return model_location
+
+
+def create_sam(model: str, device: str) -> Sam:
+    """Load the segment-anything model, fetching the model-file as necessary."""
+    model_path = get_downloaded_model_path(model)
+
+    logging.info("PyTorch version: {}".format(torch.__version__))
+    logging.info("Torchvision version: {}".format(torchvision.__version__))
+    logging.info("CUDA is available: {}".format(torch.cuda.is_available()))
+
+    logging.info("Building sam from: {}".format(model_path))
+    sam = sam_model_registry[model](checkpoint=model_path)
+    return sam.to(device=device)
+
+
+def run_segmentation(
+    predictor: SamPredictor, image: Mat, boxes_filt, prompt: str
+) -> None:
+    """Run segmentation on a single image."""
+    rr.log_image("image", image)
+    if boxes_filt.shape[0] == 0:
+        return
+    logging.info("Finding masks")
+    transformed_boxes = predictor.transform.apply_boxes_torch(
+        boxes_filt, image.shape[:2]
+    )
+
+    masks, _, _ = predictor.predict_torch(
+        point_coords=None,
+        point_labels=None,
+        boxes=transformed_boxes.to(predictor.device),
+        multimask_output=False,
+    )
+
+    logging.info("Found {} masks".format(len(masks)))
+    # mask_tensor = masks.squeeze().numpy().astype("uint8") * 128
+    # rr.log_tensor(f"query_{idx}/mask_tensor", mask_tensor)
+
+    # TODO(jleibs): we could instead draw each mask as a separate image layer, but the current layer-stacking
+    # does not produce great results.
+    masks_with_ids = list(enumerate(masks.cpu(), start=1))
+
+    # Work-around for https://github.com/rerun-io/rerun/issues/1782
+    # Make sure we have an AnnotationInfo present for every class-id used in this image
+    # TODO(jleibs): Remove when fix is released
+    rr.log_annotation_context(
+        "image",
+        [rr.AnnotationInfo(id) for id, _ in masks_with_ids],
+        timeless=False,
+    )
+
+    # Layer all of the masks together, using the id as class-id in the segmentation
+    segmentation_img = np.zeros((image.shape[0], image.shape[1]))
+    for id, m in masks_with_ids:
+        segmentation_img[m.squeeze()] = id
+
+    rr.log_segmentation_image(f"image/{prompt}/masks", segmentation_img)
```

Comment:
I don't know if this handling of masks makes as much sense for this example where we have multiple prompts. I think here it makes much more sense to have a single class id per prompt (I.e merge all masks from the same prompt). You would then set the annotation context as a timeless thing in the beginning of the script. In addition the prompt should be the label in the AnnotationInfo.